### DISCUSSION TITLE
PRONTO version 1.0

### DISCUSSION THREAD

**Discussion by xiaoliz0** [InPreD/PRONTO#4] [General]
This is PRONTO release version 1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/InPreD/PRONTO/releases/tag/v1.0'>PRONTO version 1.0</a>.</em>

**Comment by xiaoliz0** [comment]
This is PRONTO release version 1.0. Please write your feedback, comments, bug reports, suggestions here. We could discuss and make a requests list for the next version updates.

**Reply by marrip** [reply]
hi Xiaoli, thanks for making this available. I was wondering about the docker image. Maybe, we can put it into our inpred dockerhub registry instead of having to build it locally? If people want to use singularity they could also just pull it from there. Just an idea. Also I saw that you use conda to install your requirements. There are some images that might be smaller in size and have pip available which should also be suitable. Let me know if this could be relevant and we can talk about the details ☺️

**Reply by xiaoliz0** [reply]
Hi Martin, the docker image file could not be uploaded to GitHub due to the file size limitation here. But Daniel helps to share it in some other ways. We should discuss more about how to share the images files in the next meeting. :)

**Reply by marrip** [reply]
what size is the image?

**Reply by xiaoliz0** [reply]
The docker image is 667M, and the singularity image is 217M.

**Reply by marrip** [reply]
667 MB is no problem for docker hub (not the same as github), we could even set up a CI integration via gitlab actions if you want. I think it would be good to publish all inpred related images under the inpred account there so that people can easily access them :)

**Reply by tinavisnovska** [reply]
having the images shared via docker hub would be great, in principle singularity images should be pull-able from there without any extra work on the developers side. However, we should be careful not to make any proprietary data accessible by accident,...

**Reply by marrip** [reply]
agreed, one should be cautious with proprietary data. I guess anything is published on GitHub it should already be checked for that - if we build directly from GitHub it might be even safer than pushing locally build images 👍

**Reply by xiaoliz0** [reply]
Okay. Docker hub sounds a good place to put the image. I will check it. :)

**Reply by marrip** [reply]
this project is doing an automatic build and push to the registry at docker hub: https://github.com/InPreD/local_app_prepper/tree/develop
and here is the repo at docker hub:
https://hub.docker.com/repository/docker/inpred/local_app_prepper/general

**Comment by xiaoliz0** [comment]
1. Upload docker image to the InPreD Docker Hub.

**Reply by marrip** [reply]
I see you added the actions! Looking good 🎉

**Reply by xiaoliz0** [reply]
Thank you for your help Martin! It worked after some failure attempts. I will test the image and update the documentation file for it. :)

**Reply by xiaoliz0** [reply]
Done. Put the Dockerfile in the InPreD/PRONTO repository and remove the PRONTO_image folder.

**Comment by xiaoliz0** [comment]
2. CNV_overview_plots file is missing in the Testing_data folder for CL_NA12878N25.

**Reply by xiaoliz0** [reply]
Done

**Comment by xiaoliz0** [comment]
3. Image scripts read the data_path of TSOPPI from the Config/configure_PRONTO.ini file.

**Reply by xiaoliz0** [reply]
Done. The data_path will be given by docker command when run PRONTO with image. Look the README page for docker commands.

**Comment by xiaoliz0** [comment]
4. PROTO will not update the InPreD_PRONTO_metadata_tsoppi.txt file if it does not exist in the folder Out/. Print a warning message not crash.

**Reply by xiaoliz0** [reply]
Done.

**Comment by tinavisnovska** [comment]
It would be great to have the possibility to define a ppt template as a parameter of PRONTO.py script. Especially handy in situations when alternative ppt template than the default one needs to be used - like when for example the year changes and we have a need to modify the template even though want to keep the old one at hand being able to use it. Another uscase (in our case at least) are reports generated within precision medicine activities which are not included into InPreD.

**Comment by tinavisnovska** [comment]
If not now, then for the future it would be great to have ChangeLog public.

**Comment by xiaoliz0** [comment]
5. PRONTO should give a warning message when the RNA smaple ID is missing in the meta file but existing int the SOPPI results. Still generate the report without sample material ID of RNA.

**Reply by xiaoliz0** [reply]
Done.

**Comment by xiaoliz0** [comment]
6. Update with new Overview of Nomenclature and upload the file to GitHub.

**Reply by xiaoliz0** [reply]
Done. Under Log/

**Comment by xiaoliz0** [comment]
7. New PP template with some sentences changes.

**Reply by xiaoliz0** [reply]
Done.

**Comment by xiaoliz0** [comment]
8. Summary changes document from Tonje to GitHub.

**Reply by xiaoliz0** [reply]
Done. Under Log/
